Frameworks/ImageIO/CGImageDestination.mm

```diff
@@ -28,75 +885,358 @@ CGImageDestinationRef CGImageDestinationCreateWithDataConsumer(CGDataConsumerRef
+    if (!idst || !image) {
+        return;
+    }
+
+    ImageDestination* imageDestination = (ImageDestination*)idst;
+    if (imageDestination.count >= imageDestination.maxCount) {
+        NSTraceInfo(TAG, @"Max number of images in destination exceeded\n");
+        return;
+    }
+
+    ComPtr<IWICBitmapFrameEncode> imageBitmapFrame;
+    IPropertyBag2* propertyBag = NULL;
+    
+    ComPtr<IWICBitmapEncoder> imageEncoder = imageDestination.idEncoder;
+
+    // If there is no Encoder, the destination has either been already finalized or not initialized yet, so return
+    if (!imageEncoder) {
+        NSTraceInfo(TAG, @"Destination object has no Encoder\n");
+        return;
+    }
+
+    HRESULT status = imageEncoder->CreateNewFrame(&imageBitmapFrame, &propertyBag);
+    if (!SUCCEEDED(status)) {
+        NSTraceInfo(TAG, @"CreateNewFrame failed with status=%x\n", status);
+        return;
+    }
+
+    // JPEG lossy compression property
+    // Note that while the scale goes from 0.0 to 1.0 on both Windows and iOS, WIC treats 0.0 as a much worse
+    // quality than iOS. In iOS, 0.0 looks relatively similar, while in Windows the image is heavily warped.
+    if (properties && CFDictionaryContainsKey(properties, kCGImageDestinationLossyCompressionQuality) &&
+            imageDestination.type == typeJPEG) {
+        PROPBAG2 option = { 0 };
+        std::wstring quality(L"ImageQuality");
+        option.pstrName = (wchar_t*)quality.c_str();
+        VARIANT varValue;    
+        VariantInit(&varValue);
+        varValue.vt = VT_R4;
+        varValue.bVal = [(id)CFDictionaryGetValue(properties, kCGImageDestinationLossyCompressionQuality) doubleValue];
+        if (varValue.bVal >= 0.0 && varValue.bVal <= 1.0) {
+            status = propertyBag->Write(1, &option, &varValue);
+            if (!SUCCEEDED(status)) {
+                NSTraceInfo(TAG, @"Property Bag Write failed with status=%x\n", status);
+                return;
+            }
+        }
+    }
+
+    status = imageBitmapFrame->Initialize(propertyBag);
+    if (!SUCCEEDED(status)) {
+        NSTraceInfo(TAG, @"Frame Initialize failed with status=%x\n", status);
+        return;
+    }
+
+    unsigned int imageWidth = CGImageGetWidth(image);
+    unsigned int imageHeight = CGImageGetHeight(image);
+
+    // Set size and resolution of the image frame
+    status = imageBitmapFrame->SetSize(imageWidth, imageHeight);
+    if (!SUCCEEDED(status)) {
+        NSTraceInfo(TAG, @"Set Frame Size failed with status=%x\n", status);
+        return;
+    }
+
+    // TIFF on iOS will only set a resolution if both DPIWidth and DPIHeight are specified, with no default values
+    // This gets written as part of writing the frame metadata, so no else part here.
+    if (imageDestination.type != typeTIFF) {
+        // Using resolution of 72 dpi as standard for now, this seems to only affect metadata and not the image itself
+        status = imageBitmapFrame->SetResolution(72.0, 72.0);
```

Comment:
I went ahead and removed this, which leads to JPEGs giving X/Y resolutions as 0. Any image I produce on iOS has 72, and I can't seem to actually modify that number, but as we discussed letting WIC write 0 automatically probably makes more sense than hardcoding values.